### PR TITLE
[IJ Plugin] Add 'Input class constructor issue' inspection

### PR DESCRIPTION
**Pull request by BoD**
https://github.com/apollographql/apollo-kotlin/assets/372852/3ff5254c-f888-44df-b96f-0acf5bb327d8

Fix for #5374

### CODE REVIEW

File: intellij-plugin/src/main/resources/inspectionDescriptions/ApolloInputConstructorNamedArgs.html [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<html>
+<body>
+Reports invocations of input class constructors without using named parameters.
```
Comment (martinbonnin) [inline intellij-plugin/src/main/resources/inspectionDescriptions/ApolloInputConstructorNamedArgs.html:3]:
```suggestion
Reports invocations of input class constructors not using named parameters.
```

File: intellij-plugin/src/main/resources/inspectionDescriptions/ApolloInputConstructorNamedArgs.html [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+<p>It is recommended to use name parameters or builders when creating input objects to avoid issues when the schema changes.</p>
```
Comment (martinbonnin) [inline intellij-plugin/src/main/resources/inspectionDescriptions/ApolloInputConstructorNamedArgs.html:5]:
```suggestion
<p>Using named parameters makes your app robust to ordering changes in your schema.</p>
```

File: intellij-plugin/src/main/resources/inspectionDescriptions/ApolloInputConstructorNamedArgs.html [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+And creating the input object without using named parameters like this:<br>
+<pre>
+PersonInput("John", "Doe")
+</pre>
+
+If the schema changes to:<br>
+<pre>
+input PersonInput {
+  lastName: String
+  firstName: String
```
Comment (martinbonnin) [inline intellij-plugin/src/main/resources/inspectionDescriptions/ApolloInputConstructorNamedArgs.html:24]:
```suggestion
input PersonInput {
  # lastName and firstName order changed
  lastName: String
  firstName: String
```

File: intellij-plugin/src/main/resources/messages/ApolloBundle.properties [hunk 4]
```diff
@@ -148,8 +148,16 @@ inspection.oneOfInputCreation.reportText.constructor.argIsAbsent=<html><tt>@oneO
 inspection.oneOfInputCreation.reportText.builder.wrongNumberOfArgs=<html><tt>@oneOf</tt> input class builder must have exactly one field set
 inspection.oneOfInputCreation.reportText.builder.argIsNull=<html><tt>@oneOf</tt> input class argument must be non-null
 
+inspection.inputConstructorNamedArgs.displayName=Input class constructor issue
```
Comment (martinbonnin) [inline intellij-plugin/src/main/resources/messages/ApolloBundle.properties:151]:
```suggestion
inspection.inputConstructorNamedArgs.displayName=Input class constructor uses unamed arguments
```

**Review by martinbonnin — APPROVED** [review]

File: intellij-plugin/src/main/kotlin/com/apollographql/ijplugin/util/Inspections.kt [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+import com.intellij.codeInsight.daemon.impl.analysis.DaemonTooltipsUtil
+import com.intellij.codeInspection.LocalInspectionTool
+import com.intellij.codeInspection.LocalQuickFix
+import com.intellij.codeInspection.ProblemDescriptorBase
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.psi.PsiElement
+
+context(LocalInspectionTool)
+fun ProblemsHolder.registerProblem(
```
Comment (martinbonnin) [inline intellij-plugin/src/main/kotlin/com/apollographql/ijplugin/util/Inspections.kt:13]: 🤯 Do we have context receivers in the codebase now?

Comment (BoD) [reply intellij-plugin/src/main/kotlin/com/apollographql/ijplugin/util/Inspections.kt:13]: Ahaha yeah actually I've used them before in the plugin code 😅

Comment (martinbonnin) [reply intellij-plugin/src/main/kotlin/com/apollographql/ijplugin/util/Inspections.kt:13]: Plugin is probably OK but I'm not sure about the compatibility implication of those so I'd rather not use them in the libs

Comment (BoD) [reply intellij-plugin/src/main/kotlin/com/apollographql/ijplugin/util/Inspections.kt:13]: Agreed 👍 We have a bit more leeway in the plugin.

**Pull request merged**